cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -614,17 +622,41 @@ func (o *objectGraph) setSoftOwnership() {
 		}
 	}
 
+	ownerClasses := map[*node]empty{}
 	clusterClasses := o.getClusterClasses()
+
 	// Cluster that uses a ClusterClass are soft owned by that ClusterClass.
 	for _, clusterClass := range clusterClasses {
 		for _, cluster := range clusters {
 			// if the cluster uses a managed topology and uses the clusterclass
 			// set the clusterclass as a soft owner of the cluster.
-			if className, ok := cluster.additionalInfo[clusterTopologyNameKey]; ok {
+			className, hasName := cluster.additionalInfo[clusterTopologyNameKey]
+			classNamespace, hasNamespace := cluster.additionalInfo[clusterTopologyNamespaceKey]
+			isOwner := false
+
+			if hasNamespace && hasName {
+				if className == clusterClass.identity.Name && clusterClass.identity.Namespace == classNamespace {
+					cluster.addSoftOwner(clusterClass)
+					isOwner = true
+				}
+			} else if hasName {
 				if className == clusterClass.identity.Name && clusterClass.identity.Namespace == cluster.identity.Namespace {
 					cluster.addSoftOwner(clusterClass)
+					isOwner = true
 				}
 			}
+
+			if !isOwner {
+				continue
+			}
+
+			// Set shouldNotDelete if clusterClass already referenced by some cluster
+			if _, found := ownerClasses[clusterClass]; found || clusterClass.identity.Namespace != cluster.identity.Namespace {
```

Comment:
I think we should do 
```suggestion
			if _, found := ownerClasses[clusterClass]; found || clusterClass.identity.Namespace != namespace {
```
Where `namespace` is the namespace being moved

Reply:
There is also a second case to be addressed, which is when a CC is in the namespace being moved, but it is referenced by a cluster outside the ns being moved.
Note: This is trikier to address, because clusters only contains Clusters in the ns being moved, and thus it is required to list all the clusters before entering this loop.

Reply:
Should be addressed now, PTAL.